lib/agent.js

```diff
@@ -171,13 +171,14 @@ const getPath = u => u.pathname + u.search + u.hash
 
 const HttpProxyAgent = require('http-proxy-agent')
 const HttpsProxyAgent = require('https-proxy-agent')
-const SocksProxyAgent = require('socks-proxy-agent')
+const {SocksProxyAgent} = require('socks-proxy-agent')
 module.exports.getProxy = getProxy
 function getProxy (proxyUrl, opts, isHttps) {
   // our current proxy agents do not support an overridden dns lookup method, so will not
   // benefit from the dns cache
   const popts = {
     host: proxyUrl.hostname,
+    hostname: proxyUrl.hostname,
```

Comment:
In the interest of keeping *our* api surface area identical across all agents (even though socks-proxy-agent decided to use non-standard parameters), we fixed this elsewhere in this file, in the if statement that checks the protocol you can see us translate `host` to `hostname` since that's what socks-proxy-agent is expecting.  Allowing either will just lead to confusion imo.

Reply:
OK, let me think. thanks

Reply:
After my inspection, it is indeed a repeated modification, and this modification has been removed. Please review the `Pull request` again. Many thanks @wraithgar